diff -pruN usb-rndis-lite/rndis_host.c usb-rndis-lite-2.6.20/rndis_host.c
--- usb-rndis-lite/rndis_host.c 2007-02-10 20:39:03.000000000 +0100
+++ usb-rndis-lite-2.6.20/rndis_host.c  2007-02-10 20:32:29.000000000 +0100
@@ -575,7 +575,7 @@ static void rndis_unbind(struct usbnet *
        struct rndis_halt       *halt;

        /* try to clear any rndis state/activity (no i/o from stack!) */
-       halt = kcalloc(1, sizeof *halt, SLAB_KERNEL);
+       halt = kcalloc(1, sizeof *halt, GFP_KERNEL);
        if (halt) {
                halt->msg_type = RNDIS_MSG_HALT;
                halt->msg_len = ccpu2(sizeof *halt);
diff -pruN usb-rndis-lite/usbnet.c usb-rndis-lite-2.6.20/usbnet.c
--- usb-rndis-lite/usbnet.c     2007-02-10 20:39:03.000000000 +0100
+++ usb-rndis-lite-2.6.20/usbnet.c      2007-02-10 20:32:10.000000000 +0100
@@ -180,9 +180,9 @@ static int init_status (struct usbnet *d
        period = max ((int) dev->status->desc.bInterval,
                (dev->udev->speed == USB_SPEED_HIGH) ? 7 : 3);

-       buf = kmalloc (maxp, SLAB_KERNEL);
+       buf = kmalloc (maxp, GFP_KERNEL);
        if (buf) {
-               dev->interrupt = usb_alloc_urb (0, SLAB_KERNEL);
+               dev->interrupt = usb_alloc_urb (0, GFP_KERNEL);
                if (!dev->interrupt) {
                        kfree (buf);
                        return -ENOMEM;
@@ -1150,7 +1150,7 @@ usbnet_probe (struct usb_interface *udev
        skb_queue_head_init (&dev->done);
        dev->bh.func = usbnet_bh;
        dev->bh.data = (unsigned long) dev;
-       INIT_WORK (&dev->kevent, kevent, dev);
+       INIT_WORK (&dev->kevent, kevent);
        dev->delay.function = usbnet_bh;
        dev->delay.data = (unsigned long) dev;
        init_timer (&dev->delay);

