--- ipaq.c	2005-12-10 11:06:14.562622024 +0100
+++ ipaq.c-compal	2005-12-10 11:03:49.016990290 +0100
@@ -246,6 +246,8 @@
 	{ USB_DEVICE(0x04AD, 0x0301) }, /* USB Sync 0301 */
 	{ USB_DEVICE(0x04AD, 0x0302) }, /* USB Sync 0302 */
 	{ USB_DEVICE(0x04AD, 0x0303) }, /* USB Sync 0303 */
+	{ USB_DEVICE(0x04AD, 0x0306) }, /* GPS Pocket PC USB Sync */
+	{ USB_DEVICE(0x04B7, 0x0531) }, /* MyGuide 7000 XL USB Sync */
 	{ USB_DEVICE(0x04C5, 0x1058) }, /* FUJITSU USB Sync */
 	{ USB_DEVICE(0x04C5, 0x1079) }, /* FUJITSU USB Sync */
 	{ USB_DEVICE(0x04DA, 0x2500) }, /* Panasonic USB Sync */
@@ -290,6 +292,7 @@
 	{ USB_DEVICE(0x05E0, 0x2008) }, /* Symbol USB Sync 0x2008 */
 	{ USB_DEVICE(0x05E0, 0x2009) }, /* Symbol USB Sync 0x2009 */
 	{ USB_DEVICE(0x05E0, 0x200A) }, /* Symbol USB Sync 0x200A */
+	{ USB_DEVICE(0x067E, 0x1001) }, /* Intermec 700 Color Mobile Computer */
 	{ USB_DEVICE(0x067E, 0x1001) }, /* Intermec Mobile Computer */
 	{ USB_DEVICE(0x07CF, 0x2001) }, /* CASIO USB Sync 2001 */
 	{ USB_DEVICE(0x07CF, 0x2002) }, /* CASIO USB Sync 2002 */
@@ -518,6 +521,7 @@
 	{ USB_DEVICE(0x3340, 0x1326) }, /* Itautec USB Sync */
 	{ USB_DEVICE(0x3340, 0x191C) }, /* YAKUMO USB Sync */
 	{ USB_DEVICE(0x3340, 0x2326) }, /* Vobis USB Sync */
+	{ USB_DEVICE(0x3340, 0x3326) }, /* Medion USB Sync */
 	{ USB_DEVICE(0x3340, 0x3326) }, /* MEDION Winodws Moble USB Sync */
 	{ USB_DEVICE(0x3708, 0x20CE) }, /* Legend USB Sync */
 	{ USB_DEVICE(0x3708, 0x21CE) }, /* Lenovo USB Sync */
