--- ipaq-orig.c	2004-09-11 09:14:12.873599768 +0000
+++ ipaq.c	2004-09-11 09:20:54.016616760 +0000
@@ -9,6 +9,15 @@
  *	the Free Software Foundation; either version 2 of the License, or
  *	(at your option) any later version.
  *
+ * (11/09/2004) Pascal
+ *      Added vendor and product id for MiTAC Mio 168 which is also sold as
+ *      Yakumo PDA delta 300 GPS and bluemedia PDA 255.
+ *      Use modprobe ipaq ttyUSB=1 for them.
+ *
+ * (18/11/2003) Olivier
+ *      Added insmod options to specify the ttyUSB port for new iPAQ
+ *      Use modprobe ipaq ttyUSB=1 or ttyUSB=0
+ *
  * (12/12/2002) ganesh
  * 	Added support for practically all devices supported by ActiveSync
  * 	on Windows. Thanks to Wes Cilldhaire <billybobjoehenrybob@hotmail.com>.
@@ -65,12 +74,13 @@
  * Version Information
  */
 
-#define DRIVER_VERSION "v0.5"
+#define DRIVER_VERSION "v0.6"
 #define DRIVER_AUTHOR "Ganesh Varadarajan <ganesh@veritas.com>"
 #define DRIVER_DESC "USB PocketPC PDA driver"
 
 static __u16 product, vendor;
 static int debug;
+static int ttyUSB = 0;
 
 /* Function prototypes for an ipaq */
 static int  ipaq_open (struct usb_serial_port *port, struct file *filp);
@@ -548,7 +560,13 @@
 		ipaq_id_table[0].idVendor = vendor;
 		ipaq_id_table[0].idProduct = product;
 	}
+
+	ipaq_device.num_bulk_in = ttyUSB + 1;
+	ipaq_device.num_bulk_out = ttyUSB + 1;
+	ipaq_device.num_ports = ttyUSB + 1;
+
 	retval = usb_register(&ipaq_driver);
+	dbg("%s - iPAQ module configured for port ttyUSB%d", __FUNCTION__, ttyUSB);
 	if (retval)
 		goto failed_usb_register;
 		  
@@ -577,6 +595,9 @@
 module_param(debug, bool, S_IRUGO | S_IWUSR);
 MODULE_PARM_DESC(debug, "Debug enabled or not");
 
+module_param(ttyUSB, int, 0);
+MODULE_PARM_DESC(ttyUSB, "User specified ttyUSB port number");
+
 module_param(vendor, ushort, 0);
 MODULE_PARM_DESC(vendor, "User specified USB idVendor");
 
