--- ipaq-orig.c	2003-06-04 17:25:06.000000000 +0200
+++ ipaq.c	2003-11-19 10:58:58.000000000 +0100
@@ -9,6 +9,10 @@
  *	the Free Software Foundation; either version 2 of the License, or
  *	(at your option) any later version.
  *
+ * (18/11/2003) Olivier
+ *      Added insmod options to specify the ttyUSB port for new iPAQ
+ *      Use modprobe ipaq ttyUSB=1 or ttyUSB=0
+ *
  * (12/12/2002) ganesh
  * 	Added support for practically all devices supported by ActiveSync
  * 	on Windows. Thanks to Wes Cilldhaire <billybobjoehenrybob@hotmail.com>.
@@ -63,6 +67,8 @@
 	static int debug = 0;
 #endif
 
+static int ttyUSB = 0;
+
 #include "usb-serial.h"
 #include "ipaq.h"
 
@@ -72,7 +78,7 @@
  * Version Information
  */
 
-#define DRIVER_VERSION "v0.5"
+#define DRIVER_VERSION "v0.6"
 #define DRIVER_AUTHOR "Ganesh Varadarajan <ganesh@veritas.com>"
 #define DRIVER_DESC "USB PocketPC PDA driver"
 
@@ -558,14 +564,21 @@
 
 static int __init ipaq_init(void)
 {
+
 	spin_lock_init(&write_list_lock);
 	info(DRIVER_DESC " " DRIVER_VERSION);
 	if (vendor) {
 		ipaq_id_table[0].idVendor = vendor;
 		ipaq_id_table[0].idProduct = product;
 	}
+
+	ipaq_device.num_bulk_in = ttyUSB + 1;
+	ipaq_device.num_bulk_out = ttyUSB + 1;
+	ipaq_device.num_ports = ttyUSB + 1;
+
 	usb_serial_register(&ipaq_device);
 
+	dbg("%s - iPAQ module configured for port ttyUSB%d", __FUNCTION__, ttyUSB);
 	return 0;
 }
 
@@ -586,6 +599,9 @@
 MODULE_PARM(debug, "i");
 MODULE_PARM_DESC(debug, "Debug enabled or not");
 
+MODULE_PARM(ttyUSB, "i");
+MODULE_PARM_DESC(ttyUSB, "User specified ttyUSB port number");
+
 MODULE_PARM(vendor, "h");
 MODULE_PARM_DESC(vendor, "User specified USB idVendor");
 
