Index: formats/vformats-xml/xml-evolution.c
===================================================================
--- formats/vformats-xml/xml-evolution.c	(revision 1122)
+++ formats/vformats-xml/xml-evolution.c	(working copy)
@@ -35,7 +35,7 @@
 static xmlNode *handle_file_as_attribute(xmlNode *root, VFormatAttribute *attr)
 {
 	osync_trace(TRACE_INTERNAL, "Handling file_as attribute");
-	xmlNode *current = xmlNewChild(root, NULL, (xmlChar*)"FormattedName", NULL);
+	xmlNode *current = xmlNewChild(root, NULL, (xmlChar*)"FileAs", NULL);
 	osxml_node_add(current, "Content", vformat_attribute_get_nth_value(attr, 0));
 	return current;
 }
@@ -478,7 +478,7 @@
 	
 	OSyncHookTables *hooks = (OSyncHookTables *)input;
 	
-	g_hash_table_insert(hooks->attributes, "FormattedName", handle_xml_file_as_attribute);
+	g_hash_table_insert(hooks->attributes, "FileAs", handle_xml_file_as_attribute);
 	g_hash_table_insert(hooks->attributes, "Manager", handle_xml_manager_attribute);
 	g_hash_table_insert(hooks->attributes, "Assistant", handle_xml_assistant_attribute);
 	g_hash_table_insert(hooks->attributes, "Anniversary", handle_xml_anniversary_attribute);
