diff -Nur wbxml2-0.9.2-orig/src/wbxml_encoder.c wbxml2-0.9.2/src/wbxml_encoder.c
--- wbxml2-0.9.2-orig/src/wbxml_encoder.c	2006-07-11 13:47:45.000000000 +0200
+++ wbxml2-0.9.2/src/wbxml_encoder.c	2006-07-20 12:21:15.000000000 +0200
@@ -2946,7 +2946,7 @@
         /* Date and time can be encoded as OPAQUE data or as a string as specified in [ISO8601]. For now we
          * keep the string... but if someone wants to code the Date and time encoding function :-)
          */
-        /* return wbxml_encode_wv_datetime(encoder, buffer); */
+        return wbxml_encode_wv_datetime(encoder, buffer);
         break;
     case WBXML_WV_DATA_TYPE_BINARY:
         /** @todo Binary Encoding !! */
